store/tikv/gcworker/gc_worker.go

```diff
@@ -1773,6 +1789,61 @@ func (w *GCWorker) saveValueToSysTable(key, value string) error {
 	return errors.Trace(err)
 }
 
+// GC placement rules when the partitions are removed by the GC worker.
+// Placement rules cannot be removed immediately after drop table / truncate table,
+// because the tables can be flashed back or recovered.
+func (w *GCWorker) doGCPlacementRules(dr util.DelRangeTask) (pid int64, err error) {
+	// Get the job from the job history
+	var historyJob *model.Job
+	failpoint.Inject("mockHistoryJobForGC", func(v failpoint.Value) {
+		args, err1 := json.Marshal([]interface{}{kv.Key{}, []int64{int64(v.(int))}})
+		if err1 != nil {
+			return
+		}
+		historyJob = &model.Job{
+			ID:      dr.JobID,
+			Type:    model.ActionDropTable,
+			RawArgs: args,
+		}
+	})
+	if historyJob == nil {
+		err = kv.RunInNewTxn(w.store, false, func(txn kv.Transaction) error {
+			var err1 error
+			t := meta.NewMeta(txn)
+			historyJob, err1 = t.GetHistoryDDLJob(dr.JobID)
+			return err1
+		})
+		if err != nil {
+			return
+		}
+		if historyJob == nil {
+			return 0, admin.ErrDDLJobNotFound.GenWithStackByArgs(dr.JobID)
+		}
+	}
+
+	// Get the partition ID from the job and DelRangeTask.
+	switch historyJob.Type {
+	case model.ActionDropTable, model.ActionTruncateTable:
+		var physicalTableIDs []int64
+		var startKey kv.Key
+		if err = historyJob.DecodeArgs(&startKey, &physicalTableIDs); err != nil {
+			return
+		}
+		// If it's a partitioned table, then the element ID is the partition ID.
+		if len(physicalTableIDs) > 0 {
```

Comment:
when `physicalTableIDs` = 1, could it be a normal table?

Reply:
No.
https://github.com/pingcap/tidb/blob/e302200999045d81b19d08d5ec240d3438674abb/ddl/table.go#L202
https://github.com/pingcap/tidb/blob/e302200999045d81b19d08d5ec240d3438674abb/ddl/table.go#L503